pkg/reconciler/clusterimagepolicy/controller.go

```diff
@@ -17,25 +17,57 @@ package clusterimagepolicy
 import (
 	"context"
 
+	"k8s.io/client-go/tools/cache"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	configmapinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/configmap"
+	secretinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/secret"
 	"knative.dev/pkg/configmap"
 	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
 
+	"github.com/sigstore/cosign/pkg/apis/config"
 	clusterimagepolicyinformer "github.com/sigstore/cosign/pkg/client/injection/informers/cosigned/v1alpha1/clusterimagepolicy"
 	clusterimagepolicyreconciler "github.com/sigstore/cosign/pkg/client/injection/reconciler/cosigned/v1alpha1/clusterimagepolicy"
 )
 
+const SystemNamespace = "cosign-system"
+
 // NewController creates a Reconciler and returns the result of NewImpl.
 func NewController(
 	ctx context.Context,
 	cmw configmap.Watcher,
 ) *controller.Impl {
 	clusterimagepolicyInformer := clusterimagepolicyinformer.Get(ctx)
+	cmInformer := configmapinformer.Get(ctx)
+	secretInformer := secretinformer.Get(ctx)
 
-	r := &Reconciler{}
+	r := &Reconciler{
+		secretlister:    secretInformer.Lister(),
+		configmaplister: cmInformer.Lister(),
+		kubeclient:      kubeclient.Get(ctx),
+	}
 	impl := clusterimagepolicyreconciler.NewImpl(ctx, r)
-	r.Tracker = impl.Tracker
+	r.tracker = impl.Tracker
 
 	clusterimagepolicyInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
+	// When the underlying ConfigMap changes,perform a global resync on
+	// ClusterImagePolicies to make sure their state is correctly reflected
+	// in the ConfigMap. This is admittedly a bit heavy handed, but I don't
+	// really see a way around it, since if something is wrong with the
+	// ConfigMap but there are no changes to the ClusterImagePolicy, it needs
+	// to be synced.
```

Comment:
I think so long as we avoid the Patch call correctly on resync, this should be sufficiently inexpensive to re-evaluate.

Reply:
Yeah, I just jotted it down while I was thinking about it. Also, I am not sure if there are going to tons of these on a cluster anyways, and yes, hoping most of them are nops.

Reply:
It's a good call-out. 👍